src/awkward/highlevel.py

```diff
@@ -357,7 +380,7 @@ def _update_class(self):
         self.__class__ = get_array_class(self._layout, self._behavior)
 
     @property
-    def attrs(self) -> Mapping[str, Any]:
+    def attrs(self) -> Mapping | AttrsNamedAxisMapping:
```

Comment:
What we want to be able to express here is a mapping that has a few known value types, and the rest all `Mapping[T, Any]`. But, Python doesn't let us express that, so I think we need to relax this type hint, and cast at the consumer side. I.e.,
```python
named_axis = tp.cast(AxisMapping, array.attrs)
```

I recognise this will probably change a lot of code!